### DISCUSSION TITLE
allow dots in document id

### DISCUSSION THREAD

**Discussion by tacman** [meilisearch/product#799] [Feedback & Feature Proposal]
> [Formatting the document id](https://www.meilisearch.com/docs/learn/getting_started/primary_key#formatting-the-document-id)
The document id must be an integer or a string. If the id is a string, it can only contain alphanumeric characters (a-z, A-Z, 0-9), hyphens (-), and underscores (_).

I have data that is keyed by domain name, and would like use that for the document id, 

```json
{'id': 'abc.com', 'stats': {} }
```

Would you consider allowing dots in the document id?

**Comment by macraig** [comment]
Hi @tacman , we agree this would be useful. If we were to implement it, that would mean you'd need to url encode the documents if you want to use the GET route. Would that be acceptable to you?

**Comment by tacman** [comment]
Yes.  Though....why?  dots in routes are valid, they don't have to be extensions.  

Regardless, yes, having the dot in the key is more important.  And most of the time I use a library to do the get (the meili-php library).

**Reply by macraig** [reply]
> dots in routes are valid, they don't have to be extensions

Yes, you're right, my bad there. Thanks for the answer!
